tests/api/test_result.py

```diff
@@ -29,9 +30,27 @@ class TestResult:
     def test_stateless_result_raises_on_state_access(self):
         """Tests that `result.state` raises an error for a stateless result.
         """
-        result = Result([[1, 2], [3, 4]], is_stateful=False)
+        result = Result(np.array([[1, 2], [3, 4]]), is_stateful=False)
```

Comment:
How come this one was needed?

Reply:
Currently, the results object always receives the samples as a NumPy array, so the tests should be consistent and test the same.